be/src/exec/pipeline/scan/scan_operator.h

```diff
@@ -54,6 +57,43 @@ class ScanOperator : public SourceOperator {
 
     Status set_finishing(RuntimeState* state) override;
 
+    Status set_epoch_finishing(RuntimeState* state) override {
+        std::lock_guard guard(_task_mutex);
+        _detach_chunk_sources();
+        _is_epoch_finished = true;
+        return Status::OK();
+    }
+
+    bool is_epoch_finished() const override {
```

Comment:
Why change the `ScanOperator`, others would be confused if you have no knowledge of `Epoch`